x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -37,53 +41,113 @@ export const uniqueMonitorIds = (items: GetMonitorStatusResult[]): Set<string> =
     return acc;
   }, new Set<string>());
 
+const MESSAGE_AVAILABILITY_MAX = 3;
+
+export const availabilityMessage = (
+  availabilityResult: GetMonitorAvailabilityResult[],
+  threshold: string
+): string => {
+  return availabilityResult.length > 0
+    ? availabilityResult
+        .sort((a, b) => (a.availabilityRatio ?? 100) - (b.availabilityRatio ?? 100))
+        .slice(0, MESSAGE_AVAILABILITY_MAX)
+        .map(({ availabilityRatio, name, monitorId, url }) =>
+          i18n.translate('xpack.uptime.alerts.availability.monitorSummary', {
+            defaultMessage: '{nameOrId}({url}): {availabilityRatio}%',
+            values: {
+              nameOrId: name || monitorId,
+              url,
+              availabilityRatio: ((availabilityRatio ?? 1.0) * 100).toPrecision(5),
+            },
+          })
+        )
+        .reduce((prev: string, cur: string, _ind: number, array: string[]) => {
```

Comment:
I'm not really sure about reduce here. We have this awkward thing where we do checks for how many things have been found previously. It seems to me that it'd be easier just to store the `map` result in a variable and use `[].join` where appropriate